src/__tests__/Batcher.ts

```diff
@@ -0,0 +1,209 @@
+import { AvoBatcher } from "../AvoBatcher";
+import { AvoInspector } from "../AvoInspector";
+import { AvoNetworkCallsHandler } from "../AvoNetworkCallsHandler";
+import { AvoStorage } from "../AvoStorage";
+
+import { defaultOptions, networkCallType } from "./constants";
+
+const inspectorVersion = process.env.npm_package_version || "";
+
+describe("Batcher", () => {
+  let checkBatchSpy: jest.SpyInstance<any, unknown[]>;
+  let inspectorCallSpy: jest.SpyInstance<any, unknown[]>;
+
+  const { apiKey, env, version } = defaultOptions;
+
+  const storage = new AvoStorage();
+  const networkHandler = new AvoNetworkCallsHandler(
+    apiKey,
+    env,
+    "",
+    version,
+    inspectorVersion,
+  );
+
+  beforeAll(() => {
+    checkBatchSpy = jest.spyOn(
+      AvoBatcher.prototype as any,
+      "checkIfBatchNeedsToBeSent",
+    );
+
+    inspectorCallSpy = jest.spyOn(
+      AvoNetworkCallsHandler.prototype as any,
+      "callInspectorWithBatchBody",
+    );
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+    AvoInspector.avoStorage.removeItem(AvoBatcher.cacheKey);
+  });
+
+  test("handleSessionStarted adds event to storage", () => {
+    const inspector = new AvoInspector(defaultOptions);
+
+    inspector.avoBatcher.handleSessionStarted();
+
+    const events = AvoInspector.avoStorage.getItem(AvoBatcher.cacheKey);
+
+    expect(events).not.toBeNull();
+
+    // FIXME: expects Object or null
+    // @ts-ignore
+    expect(events.length).toEqual(1);
+
+    // FIXME: expects Object or null
```

Comment:
What is the problem here?

Reply:
It expect that `getItem()` method returns object or null, but here the events is returned as an array

Reply:
I think it is designed to be an array, so you can put multiple session/schema payloads under the key and can access them all.

That's probably another test case, put a mix of sessions and schema payloads and see if all can be accessed from the storage.

Reply:
and also there is a limit on saved events, if you try to put more than 1000 events oldest events should be dropped and only 1000 latest events should be in the storage

Reply:
> I think it is designed to be an array, so you can put multiple session/schema payloads under the key and can access them all.
> 
> That's probably another test case, put a mix of sessions and schema payloads and see if all can be accessed from the storage.

`getItem()` implementation defines that returned value is an object or null, but events are stored in storage as an array, yes. Will look mere into this 👍  For sessions/events there was a test case, where multiple events are stored in storage and then retrieved.

Reply:
> and also there is a limit on saved events, if you try to put more than 1000 events oldest events should be dropped and only 1000 latest events should be in the storage

Missed this limitation, will add tests for this 👍